src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/Input/KeyConverter.cs

```diff
@@ -4,246 +4,326 @@
-                        case "RIGHTWINDOWS": keyFound = Key.RWin; break;
-                        case "APPS": keyFound = Key.Apps; break;
-                        case "APPLICATION": keyFound = Key.Apps; break;
-                        case "BREAK": keyFound = Key.Cancel; break;
-                        case "BACKSPACE": keyFound = Key.Back; break;
-                        case "BKSP": keyFound = Key.Back; break;
-                        case "BS": keyFound = Key.Back; break;
-                        case "SHIFT": keyFound = Key.LeftShift; break;
-                        case "LEFTSHIFT": keyFound = Key.LeftShift; break;
-                        case "RIGHTSHIFT": keyFound = Key.RightShift; break;
-                        case "CONTROL": keyFound = Key.LeftCtrl; break;
-                        case "CTRL": keyFound = Key.LeftCtrl; break;
-                        case "LEFTCTRL": keyFound = Key.LeftCtrl; break;
-                        case "RIGHTCTRL": keyFound = Key.RightCtrl; break;
-                        case "ALT": keyFound = Key.LeftAlt; break;
-                        case "LEFTALT": keyFound = Key.LeftAlt; break;
-                        case "RIGHTALT": keyFound = Key.RightAlt; break;
-                        case "SEMICOLON": keyFound = Key.OemSemicolon; break;
-                        case "PLUS": keyFound = Key.OemPlus; break;
-                        case "COMMA": keyFound = Key.OemComma; break;
-                        case "MINUS": keyFound = Key.OemMinus; break;
-                        case "PERIOD": keyFound = Key.OemPeriod; break;
-                        case "QUESTION": keyFound = Key.OemQuestion; break;
-                        case "TILDE": keyFound = Key.OemTilde; break;
-                        case "OPENBRACKETS": keyFound = Key.OemOpenBrackets; break;
-                        case "PIPE": keyFound = Key.OemPipe; break;
-                        case "CLOSEBRACKETS": keyFound = Key.OemCloseBrackets; break;
-                        case "QUOTES": keyFound = Key.OemQuotes; break;
-                        case "BACKSLASH": keyFound = Key.OemBackslash; break;
-                        case "FINISH": keyFound = Key.OemFinish; break;
-                        case "ATTN": keyFound = Key.Attn; break;
-                        case "CRSEL": keyFound = Key.CrSel; break;
-                        case "EXSEL": keyFound = Key.ExSel; break;
-                        case "ERASEEOF": keyFound = Key.EraseEof; break;
-                        case "PLAY": keyFound = Key.Play; break;
-                        case "ZOOM": keyFound = Key.Zoom; break;
-                        case "PA1": keyFound = Key.Pa1; break;
-                        default: keyFound = Enum.Parse<Key>(keyToken, true); break;
-                    }
-
-                    if ((int)keyFound != -1)
-                    {
-                        return keyFound;
-                    }
-                    return null;
```

Comment:
This is a deadcode, as a result, the block `throw new NotSupportedException(SR.Format(SR.Unsupported_Key, fullName))` in `ConvertFrom` would never be thrown and instead we get `ArgumentException` from `Enum.Parse<Key>`, however I think we should adjust the behavior using `Enum<Key>.TryParse` to throw this on `false` as it seemed intended and correct behavior with other converters.

Opinions?

Reply:
I think it would be a compatibility concern to throw a different type of exception. `ArgumentException` seems appropriate anyway...

Reply:
@miloush Other converters throw `NotSupportedException` with appropriate strings, e.g. `Unsupported_Modifier` in `ModifierKeysConverter` or `Unsupported_MouseAction` in `MouseActionConverter`.

Reply:
An argument for aligning is that `NotSupportedException` is the documented one: https://learn.microsoft.com/en-us/dotnet/api/system.windows.input.keyconverter.convertfrom

Reply:
Yeah, I'm inclined to align this one but maybe in a different PR since I wanna go over gestures init and conversion which is fugly, which requires me to touch the converter with an internal method to do it cleanly with spans and without duplicating functionality.

Reply:
I'll create a separate issue and PR for this particular case, can fix merge conflicts later.